Sources/NIOTestUtils/ByteToMessageDecoderVerifier.swift

```diff
@@ -0,0 +1,211 @@
+                    }
+                    for expectedOutput in expectedOutputs {
+                        guard let actualOutput = try channel.readInbound(as: Decoder.InboundOut.self) else {
+                            throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                         errorCode: .noOutputResult(expectedOutput))
+                        }
+                        guard actualOutput == expectedOutput else {
+                            throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                         errorCode: .wrongDecodingResult(actual: actualOutput,
+                                                                                         expected: expectedOutput))
+                        }
+                    }
+                    let actualExtraOutput = try channel.readInbound(as: Decoder.InboundOut.self)
+                    guard actualExtraOutput == nil else {
+                        throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                     errorCode: .unexpectedDecodingResult(actualExtraOutput!))
+                    }
+                }
+            }
+        }
+
+        func verifyManyAtOnce(channel: RecordingChannel) throws {
+            var overallBuffer = channel.allocator.buffer(capacity: 1024)
+            var overallExpecteds: [Out] = []
+
+            for _ in 0..<10 {
+                for (var input, expectedOutputs) in inputOutputPairs.shuffled() {
+                    overallBuffer.writeBuffer(&input)
+                    overallExpecteds.append(contentsOf: expectedOutputs)
+                }
+            }
+
+            try channel.writeInbound(overallBuffer)
+            for expectedOutput in overallExpecteds {
+                guard let actualOutput = try channel.readInbound(as: Decoder.InboundOut.self) else {
+                    throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                 errorCode: .noOutputResult(expectedOutput))
+                }
+                guard actualOutput == expectedOutput else {
+                    throw VerificationError<Out>(inputs: channel.inboundWrites,
+                                                 errorCode: .wrongDecodingResult(actual: actualOutput,
+                                                                                 expected: expectedOutput))
+                }
+            }
+        }
+
+        let decoder: Decoder = decoderFactory()
+        let channel = RecordingChannel(EmbeddedChannel(handler: ByteToMessageHandler<Decoder>(decoder)))
+
+        try verifySimple(channel: channel)
+        try verifyDripFeed(channel: channel)
+        try verifyManyAtOnce(channel: channel)
+
+        if case .leftOvers(inbound: let ib, outbound: let ob, pendingOutbound: let pob) =  try channel.finish() {
```

Comment:
Super duper nit: delete one space character before `try`. Sorry, if I wasn't already leaving feedback I'd have let this one slide. 😆

Reply:
done